server/src/core/user/user-service.ts

```diff
@@ -135,6 +135,16 @@ export class Users implements UserService {
     }
   }
 
+  async getAllBeneficiariesByUser(userId: string): Promise<User[]> {
+    try {
+      const result = await this.collection.find({ donors: { $in: [userId] } }).toArray();
+      return result;
+    }
+    catch (e) {
+      throw createDbOpFailedError(e.message);
+    }
+  }
+
```

Comment:
This result includes sensitive data. All queries that return users should apply the `SAFE_USER_PROJECTION`, which removes sensitive data like the hashed password from the result.